rocketpy/AeroSurface.py

```diff
@@ -368,11 +313,94 @@ def evaluate_geometrical_parameters(self):
+
+        # Calculate a function to find the tangential intersection point between the circle and nosecone curve.
+        def find_x_intercept(x):
+            return x + self.y_nosecone(x) * self.y_nosecone.differentiate(x)
+
+        # Calculate a function to find the radius of the nosecone curve
+        def find_radius(x):
+            return (self.y_nosecone(x) ** 2 + (x - find_x_intercept(x)) ** 2) ** 0.5
+
+        # Check bluffness circle and choose whether to use it or not
+        if self.bluffness is None or self.bluffness == 0:
+            # Set up parameters to continue without bluffness
+            r_circle, circle_center, x_init = 0, 0, 0
+        else:
+            # Calculate circle radius
+            r_circle = self.bluffness * self.base_radius
+            if self.kind == "elliptical":
+                # Calculate a function to set up a circle at the starting position to test bluffness
+                def test_circle(x):
+                    return np.sqrt(r_circle**2 - (x - r_circle) ** 2)
+
+                # Check if bluffness circle is too small
+                if test_circle(1e-03) <= self.y_nosecone(1e-03):
+                    # Raise a warning
+                    warnings.warn(
+                        "WARNING: The chosen bluffness ratio is too small for "
+                        "the selected nose cone category, thereby the effective "
+                        "bluffness will be 0."
+                    )
+                    # Set up parameters to continue without bluffness
+                    r_circle, circle_center, x_init = 0, 0, 0
+                else:
+                    # Find the intersection point between circle and nosecone curve
+                    x_init = fsolve(lambda x: find_radius(x) - r_circle, r_circle)[0]
+                    circle_center = find_x_intercept(x_init)
+            else:
+                # Find the intersection point between circle and nosecone curve
+                x_init = fsolve(lambda x: find_radius(x) - r_circle, r_circle)[0]
+                circle_center = find_x_intercept(x_init)
+
+        # Calculate a function to create the circle at the correct position
+        def create_circle(x):
+            return abs(r_circle**2 - (x - circle_center) ** 2) ** 0.5
+
+        # Define a function for the final shape of the curve with a circle at the tip
+        def final_shape(x):
+            return self.y_nosecone(x) if x >= x_init else create_circle(x)
+
+        # Vectorize the final_shape function
+        final_shape_vec = np.vectorize(final_shape)
+
+        # Create the vectors X and Y with the points of the curve
+        self.nosecone_x = (self.length - (circle_center - r_circle)) * (
+            np.linspace(0, 1, n) ** p
+        )
+        self.nosecone_y = final_shape_vec(self.nosecone_x + (circle_center - r_circle))
+
+        # Evaluate final geometry parameters
+        self.length = self.nosecone_x[-1]
```

Comment:
This method is modifying the length of the NoseCone, right?

Should this affect the position of the C.P.? 

Imagine you create a rocket with a quite long nosecone,
Then you run the draw method, which calls the `evaluate_nose_shape` method, the length will be modified and therefore the flight simulation would have a different behavior.

I'm keeping this short, but try to clearly define if the method actually modifies the CP or not

Reply:
Yeah, this was an overlook. Had to change a bit of code to take into account this length change when there is bluffness

Reply:
Solved in https://github.com/RocketPy-Team/RocketPy/pull/413/commits/e847406d9df8eb36befc3f1d33fb9e46c25f00e4